models/people.py

```diff
@@ -0,0 +1,62 @@
+class Person(object):
+    """
+    Creates a Person object. Classes Fellow and Staff inherit from it.
+    """
+    def __init__(self, name):
+        self.name = name
+        self.emp_id = id(self)
+
+    def __repr__(self):
+        return "<Room %s>" % self.name
+
+
+class Staff(Person):
+
+    job_type = "Staff"
+
+    def __init__(self, name):
+        super(Staff, self).__init__(name)
+        self.emp_id = id(self)
+
+    def __repr__(self):
+        return "<Staff %s>" % self.name
+
+
+class Fellow(Person):
+
+    job_type = "Fellow"
+
+    def __init__(self, name):
+        super(Fellow, self).__init__(name)
+        self.emp_id = id(self)
+
+    def __repr__(self):
+        return "Fellow %s>" % self.name
+
+
+class StaffFromDatabase(Person):
```

Comment:
The reason for this was because the `Staff` class uses Python's `id()` function for the emp_id field. This wouldn't be ideal if loading people from the database, because they'd lose their emp_id. This would in turn mess up with the allocations data, since emp_id is the foreign key used to link allocated people with the room they've been allocated to. So I used the `StaffFromDatabase` class where emp_id is picked from the database as opposed to being created.